rust/tw_keypair/src/ed25519/modifications/cardano/extended_keypair.rs

```diff
@@ -35,8 +35,10 @@ impl<H: Hasher512> SigningKeyTrait for ExtendedKeyPair<H> {
     type Signature = Signature;
 
     fn sign(&self, message: Self::SigningMessage) -> KeyPairResult<Self::Signature> {
-        self.private()
-            .sign_with_public_key(self.public(), message.as_slice())
+        unsafe {
```

Comment:
I would like to point out that the term 'unsafe' in Rust pertains to memory safety guarantees, while the Chalkias attack does not relate to memory safety. Thank you for considering this distinction.

Reply:
Personally, I think it's ok to use unsafe, but i want to follow best practices, so thank you for advising me.
I removed `unsafe` keyword and renamed the method to `dangerous_sign_with_pubkey`